src/hyperion/external_interaction/callbacks/xray_centre/nexus_callback.py

```diff
@@ -55,12 +59,56 @@ def descriptor(self, doc):
 
             LOGGER.info("Initialising nexus writers")
             nexus_data_1 = self.parameters.get_nexus_info(1)
+            LOGGER.info(f"Nexus data 1: {nexus_data_1}")
             nexus_data_2 = self.parameters.get_nexus_info(2)
-            self.nexus_writer_1 = NexusWriter(self.parameters, **nexus_data_1)
+            self.nexus_writer_1 = NexusWriter(
+                self.parameters,
+                create_goniometer_func=self.create_goniometer_func,
+                **nexus_data_1,
+            )
             self.nexus_writer_2 = NexusWriter(
                 self.parameters,
+                create_goniometer_func=self.create_goniometer_func,
                 **nexus_data_2,
                 vds_start_index=nexus_data_1["data_shape"][0],
             )
             self.nexus_writer_1.create_nexus_file()
             self.nexus_writer_2.create_nexus_file()
+
+
+class Gridscan2DNexusFileCallback(CallbackBase):
+    """Similar to above, but for a 2D gridscan"""
```

Comment:
I think the duplication with the class above shouldn't block merging in the short term, but let's write an issue to improve this in the longer term.